### PR TITLE
Add Django 4.1 support

### PR DESCRIPTION
**Pull request by dmkoch**
[//]: # (Thanks for helping us out! Cache invalidation sucks, so your help is greatly appreciated!)

## Description

Add Django 4.1 support to CI

## Rationale

Django 4.1 is the current version of Django

### CODE REVIEW

**Comment by Andrew-Chen-Wang** [comment]
thanks @dmkoch ! why was the 4.0 specific change required?

**Comment by dmkoch** [comment]
> thanks @dmkoch ! why was the 4.0 specific change required?

@Andrew-Chen-Wang without the change this test was failing:

```
======================================================================
FAIL: test_subquery (cachalot.tests.read.ReadTestCase) [CACHALOT_FINAL_SQL_CHECK = True]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/workspaces/django-cachalot/cachalot/tests/tests_decorators.py", line 23, in wrapper
    func(self, *args, **kwargs)
  File "/workspaces/django-cachalot/cachalot/tests/read.py", line 365, in test_subquery
    self.assert_tables(
  File "/workspaces/django-cachalot/cachalot/tests/test_utils.py", line 39, in assert_tables
    self.assertSetEqual(_get_tables(queryset.db, queryset.query), tables, str(queryset.query))
AssertionError: Items in the second set but not the first:
'django_content_type' : SELECT "cachalot_test"."id", "cachalot_test"."name", "cachalot_test"."owner_id", "cachalot_test"."public", "cachalot_test"."date", "cachalot_test"."datetime", "cachalot_test"."permission_id", "cachalot_test"."a_float", "cachalot_test"."a_decimal", "cachalot_test"."bin", "cachalot_test"."ip", "cachalot_test"."duration", "cachalot_test"."uuid", "cachalot_test"."json" FROM "cachalot_test" INNER JOIN "auth_user" ON ("cachalot_test"."owner_id" = "auth_user"."id") INNER JOIN "auth_user_groups" ON ("auth_user"."id" = "auth_user_groups"."user_id") INNER JOIN "auth_group" ON ("auth_user_groups"."group_id" = "auth_group"."id") INNER JOIN "auth_group_permissions" ON ("auth_group"."id" = "auth_group_permissions"."group_id") WHERE "auth_group_permissions"."permission_id" IN (SELECT "auth_permission"."id" FROM "auth_permission") ORDER BY "cachalot_test"."name" ASC
```

File: cachalot/tests/read.py [hunk 1]
```diff
@@ -352,7 +352,7 @@ def test_many_to_many_when_no_sql_check(self):
     @all_final_sql_checks
     def test_subquery(self):
         additional_tables = []
-        if django_version[0] >= 4 and settings.CACHALOT_FINAL_SQL_CHECK:
+        if django_version[0] >= 4 and django_version[1] < 1 and settings.CACHALOT_FINAL_SQL_CHECK:
```
Comment (Andrew-Chen-Wang) [inline cachalot/tests/read.py]: this would also pass for django 5.0

Comment (dmkoch) [reply cachalot/tests/read.py]: I changed it so it's looking for 4.0.x only

**Comment by Andrew-Chen-Wang** [comment]
lgtm thanks again! open a pr and will make the release once i'm back on a computer

**Comment by dmkoch** [comment]
@Andrew-Chen-Wang Here is the PR: https://github.com/noripyt/django-cachalot/pull/219

**Pull request closed without merging**